qiskit/providers/aer/primitives/estimator.py

```diff
@@ -0,0 +1,307 @@
+        backend_options: dict | None = None,
+        transpile_options: dict | None = None,
+        approximation: bool = False,
+    ):
+        """
+        Args:
+            circuits: Quantum circuits that represent quantum states.
+            observables: Observables.
+            parameters: Parameters of quantum circuits, specifying the order in which values
+                will be bound. Defaults to ``[circ.parameters for circ in circuits]``
+                The indexing is such that ``parameters[i, j]`` is the j-th formal parameter of
+                ``circuits[i]``.
+            backend_options: Options passed to AerSimulator.
+            transpile_options: Options passed to transpile.
+            approximation: if True, it calculates expectation values with normal distribution
+                approximation.
+        """
+        if isinstance(circuits, QuantumCircuit):
+            circuits = (circuits,)
+        circuits = tuple(init_circuit(circuit) for circuit in circuits)
+
+        if isinstance(observables, (PauliSumOp, BaseOperator)):
+            observables = (observables,)
+        observables = tuple(
+            init_observable(observable).simplify(atol=0) for observable in observables
+        )
+
+        super().__init__(
+            circuits=circuits,
+            observables=observables,
+            parameters=parameters,
+        )
+        self._is_closed = False
+        backend_options = {} if backend_options is None else backend_options
+        method = (
+            "density_matrix"
+            if approximation and "noise_model" in backend_options
+            else "statevector"
+        )
+        self._backend = AerSimulator(method=method)
+        self._backend.set_options(**backend_options)
+        self._transpile_options = Options()
+        if transpile_options is not None:
+            self._transpile_options.update_options(**transpile_options)
+        self.approximation = approximation
+
+    def _call(
+        self,
+        circuits: Sequence[int],
+        observables: Sequence[int],
+        parameter_values: Sequence[Sequence[float]],
+        **run_options,
+    ) -> EstimatorResult:
+        if self._is_closed:
+            raise QiskitError("The primitive has been closed.")
+
+        seed = run_options.pop("seed", None)
+        if seed is not None:
+            run_options.setdefault("seed_simulator", seed)
+
+        if self.approximation:
+            shots = run_options.pop("shots", None)
+
+            experiments = []
+            parameter_binds = []
+            for i, j, value in zip(circuits, observables, parameter_values):
+                if len(value) != len(self._parameters[i]):
+                    raise QiskitError(
+                        f"The number of values ({len(value)}) does not match "
+                        f"the number of parameters ({len(self._parameters[i])})."
+                    )
+
+                circuit = self._circuits[i].copy()
```

Comment:
can we avoid copy by calling pop of ciruict.data at the end?

Reply:
I think it would be difficult because the same `I` could be given.